collector/softnet_linux.go

```diff
@@ -65,19 +69,45 @@ func NewSoftnetCollector(logger log.Logger) (Collector, error) {
 			"Number of times processing packets ran out of quota",
 			[]string{"cpu"}, nil,
 		),
+		cpuCollision: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, softnetSubsystem, "cpu_collision_total"),
+			"Number of collision occur while obtaining device lock while transmitting",
+			[]string{"cpu"}, nil,
+		),
+		receivedRps: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, softnetSubsystem, "received_rps_total"),
+			"Number of times cpu woken up received_rps",
+			[]string{"cpu"}, nil,
+		),
+		flowLimitCount: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, softnetSubsystem, "flow_limit_count_total"),
+			"Number of times flow limit has been reached",
+			[]string{"cpu"}, nil,
+		),
+		softnetBacklogLen: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, softnetSubsystem, "backlog_len"),
+			"Softnet backlog status",
+			[]string{"cpu"}, nil,
+		),
 		logger: logger,
 	}, nil
 }
 
 // Update gets parsed softnet statistics using procfs.
 func (c *softnetCollector) Update(ch chan<- prometheus.Metric) error {
+	var cpu string
+
 	stats, err := c.fs.NetSoftnetStat()
 	if err != nil {
 		return fmt.Errorf("could not get softnet statistics: %w", err)
 	}
 
 	for cpuNumber, cpuStats := range stats {
-		cpu := strconv.Itoa(cpuNumber)
+		if cpuStats.Width >= 13 {
+			cpu = strconv.FormatUint(uint64(cpuStats.Index), 10)
+		} else {
+			cpu = strconv.Itoa(cpuNumber)
+		}
```

Comment:
I'll leave it up to @discordianfish to decide if we want to do this, or the procfs change.

Reply:
It's fine to do this here for now but let's add a code comment to give some context here.

Reply:
I would rather control this logic in procfs. I've updated the procfs library to know about the kernel versions. It should be safe to use `cpuStats.Index` without any special handling now.

Reply:
done